### PR TITLE
Issue #8828: G now also gives real solutions

### PR DESCRIPTION
**Pull request by mihir-w**
Fixed formatting of verification of solution at the end. Removed one instance of `bad_result` in the code.

Although I do still think that `got_s` logically should be inside the for loop, but since everything seems to working fine for now, I have kept it outside. Will try to come up with a test case later.

One of the test case function `test_issue_5849()` has solutions which do not satisfy the equations. Substitution of the given answer does not result in zero and hence `solve` was getting the required answer but returned the empty list since verification failed. So I changed the test case.

Sorry for the mess, in the last [PR](https://github.com/sympy/sympy/pull/8869). Have cherry-picked the commits from the last pull request.

Thanks for all the help!
#8828

### CODE REVIEW

File: sympy/solvers/solvers.py [hunk 1]
```diff
@@ -1546,7 +1551,11 @@ def _expand(p):
                     rv = set([inv.subs(isym, xi) for xi in _solve(eq, isym, **flags)])
                 else:
                     try:
+<<<<<<< HEAD
                         rv = set(_solve(eq, symbol, **flags))
```
Comment (smichr) [inline sympy/solvers/solvers.py]: keep this version with "set"

File: sympy/solvers/solvers.py [hunk 2]
```diff
@@ -767,8 +767,13 @@ def solve(f, *symbols, **flags):
     Although eq or eq1 could have been used to find xvals, the solution can
     only be verified with expr1:
 
+<<<<<<< HEAD
         >>> z = expr - v
```
Comment (smichr) [inline sympy/solvers/solvers.py]: keep these first two lines

**Comment by smichr** [comment]
[New PR](https://github.com/mihir-w/sympy/pull/3).

**Comment by smichr** [comment]
BTW, there is no need to have closed the old commit: just make local changes and then do 

`git push -f mihir-w yourbranch`

The "-f' will force the PR to update; but if you make changes while a PR is closed you won't be able to reopen it. (So in this case I don't think you could go back to the old PR.)

**Comment by mihir-w** [comment]
Oh! I saw the PR just after I made the commit regarding merge conflicts. 
Will do the remaining changes.

**Comment by smichr** [comment]
I just updated the PR so it should go in ok...eventually this will all be flattened to a single commit.

**Comment by mihir-w** [comment]
I am also seeing changes in elementary.py. Are they part of this PR or some other commit which my branch is not updated with?

I did `git pull --rebase upstream master` followed by `git rebase master` in my branch but my elementary.py file remains unchanged.

**Comment by smichr** [comment]
oh, sorry...I'm looking at several PRs right now and that is a bleed-through from one of them. Please disregard the elementary.py changes.

**Comment by mihir-w** [comment]
Just added the required changes. Regarding squashing of commits, should I squash all the commits into one ?

**Comment by smichr** [comment]
> old commits are coming up again

This can happen if you do an interactive rebase on something that has been merged with master. What I do in these situations is "git merge master;git diff master > diff; git checkout master;git checkout -b tmp;git apply diff".

> squash to one commit

Yes, that's fine.

**Comment by mihir-w** [comment]
@smichr done.

**Comment by mihir-w** [comment]
Two tests are primarily failing.

First:
```
 p, q, r = [set([tuple(i.evalf(2) for i in j) for j in R]) for R in A, B, C]
                                                                        ^
SyntaxError: invalid syntax
Traceback (most recent call last):
  File "<stdin>", line 5, in <module>
Exception: Tests failed
```
Shows invalid syntax for python versions >= 3.2

Second:
```
_____________________ sympy.diffgeom.diffgeom.CoordSystem ______________________
File "/home/travis/virtualenv/pypy-2.4.0/site-packages/sympy-0.7.6_git-py2.7.egg/sympy/diffgeom/diffgeom.py", line 118, in sympy.diffgeom.diffgeom.CoordSystem
Failed example:
    rect.coord_tuple_transform_to(polar, [1, 1])
Expected:
    Matrix([
    [sqrt(2)],
    [   pi/4]])
Got:
    Matrix([
    [(-2 + sqrt(2))/(-sqrt(2) + 1)],
    [                         pi/4]])
```
Above test fails primarily when `FASTCACHE=False` and once when `TEST_ASCII=True`

First error seems easy enough, but I don't know why exactly second is failing. Why is `FASTCACHE=False` causing problem ?

**Comment by smichr** [comment]
The second test makes it look like simplification did not occur since the value shown there simplifies to sqrt(2). I don't know why this would depend on the settings of FASTCACHE. Maybe @pbrady or @asmeurer would have some ideas.

**Comment by pbrady** [comment]
In order to test all of our caching systems, we don't use fastcache for doctests.  

Do the errors actually depend on which cache is used (`FASTCACHE=False` triggers the use of the pure python cache)?  Do the doctests pass if we use fastcache?

My hunch is that the pure python version and fastcache will yield the same result (if they don't then that is a bug in fastcache).

**Comment by smichr** [comment]
A workaround (if the cause is not easily determined) is to just make the line read 

```
rect.coord_tuple_transform_to(polar, [1, 1]).applyfunc(simplify)
```

**Comment by mihir-w** [comment]
`simplify` in the final loop is changing the result.
Before applying `simplify`:
```
print(result)
[{_theta: -2*atan((_r + sqrt(_r**2 + _theta**2))/_theta), _r: -_theta/sin(2*atan((_r + sqrt(_r**2 + _theta**2))/_theta))}, {_theta: -2*atan((_r - sqrt(_r**2 + _theta**2))/_theta), _r: -_theta/sin(2*atan((_r - sqrt(_r**2 + _theta**2))/_theta))}]
```

After simplification:
```
[{_theta: -2*atan((_r + sqrt(_r**2 + _theta**2))/_theta), _r: -(_r**2 + _r*sqrt(_r**2 + _theta**2) + _theta**2)/(_r + sqrt(_r**2 + _theta**2))}, {_theta: -2*atan((_r - sqrt(_r**2 + _theta**2))/_theta), _r: (-_r**2 + _r*sqrt(_r**2 + _theta**2) - _theta**2)/(_r - sqrt(_r**2 + _theta**2))}]
```

Values of `_r` are different, which cause the change at the end.

**Comment by mihir-w** [comment]
I guess changing the line to what you said is probably the best solution. ( Minimum code change )
@smichr should I go ahead ?

**Comment by smichr** [comment]
I would recommend it. It would be nice to fix the error, but that should really be a separate issue.

**Comment by mihir-w** [comment]
@smichr I guess this can now be merged ?

**Comment by smichr** [comment]
It is being merged. Thanks for working on this issue!

**Comment by mihir-w** [comment]
Thanks for your help. Learned a lot!

**Comment by smichr** [comment]
p.s. `got_s` does need to go inside the loop

**Pull request merged**